src/bridges/uniswapv3/SyncUniswapV3Bridge.sol

```diff
@@ -0,0 +1,449 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Aztec.
+pragma solidity >=0.8.4;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {AztecTypes} from "../../aztec/libraries/AztecTypes.sol";
+import {UniswapV3Bridge, TransferHelper, ISwapRouter} from "./UniswapV3Bridge.sol";
+import {IUniswapV3Pool} from "./interfaces/IUniswapV3Pool.sol";
+import {IDefiBridge} from "../../aztec/interfaces/IDefiBridge.sol";
+import {console} from "../../../lib/forge-std/src/console.sol";
+
+contract SyncUniswapV3Bridge is IDefiBridge, UniswapV3Bridge {
+    using SafeERC20 for IERC20;
+
+    /*
+        STRUCTS AND ENUMS
+    */
+
+    //used as a record for MINT_PT1 & MINT_P2
+    struct MintFunding {
+        address token;
+        uint256 amount;
+    }
+
+    error InvalidCaller();
+    error InvalidOutputs();
+    error InvalidRefund();
+    /* 
+        IMMUTABLES
+    */
+
+    /* 
+        MUTABLE VARIABLES
+    */
+
+    mapping(uint256 => MintFunding) public syncMintFundingMap; //interaction nonce -> MintFunding struct for MINT_PT1 interactions
+
+    modifier onlyRollup() {
+        if (msg.sender != address(ROLLUP_PROCESSOR)) revert InvalidCaller();
+        _;
+    }
+
+    constructor(
+        address _rollupProcessor,
+        address _router,
+        address _nonfungiblePositionManager,
+        address _factory,
+        address _wEth
+    ) public UniswapV3Bridge(_rollupProcessor, _router, _nonfungiblePositionManager, _factory, _wEth) {}
+
+    /**
+     * @notice  packs _auxData for front end user
+     * @dev The first 24 bits are tickLower. The second 24 are tickUpper. The last 16 the pool's fee.
+     * As a pool's fee ranges from 10 bps, 100 bps, 300 bps, and 1000 bps, there is no data loss and type conversion
+     * is acceptable.
+     * @param _tickLower the lower range of the position
+     * @param _tickUpper the upper range of the position
+     * @param _fee the fee tier of the pool
+     * @return _auxData the packed _auxData
+     */
+
+    function packData(
```

Comment:
Please remove the blank line between function definition and natspec everywhere.